packages/realm/src/Logger.ts

```diff
@@ -40,7 +159,30 @@ export enum NumericLogLevel {
  */
 export type Logger = (level: NumericLogLevel, message: string) => void;
 
-export type LoggerCallback = (level: LogLevel, message: string) => void;
+/**
+ * A callback passed to `Realm.setLogger`.
+ *
+ * @param level   - The level of the log entry.
+ * @param message - The message of the log entry.
+ * @since 12.0.0
+ * @deprecated Will be removed in v13.0.0
+ */
+export type LoggerCallback1 = (level: LogLevel, message: string) => void;
+export type LoggerCallbackArgs = {
+  category: LogCategory;
+  level: LogLevel;
+  message: string;
+};
+/**
+ * A callback passed to `Realm.setLogger`. Arguments are passed as a POJO.
+ *
+ * @param category   - The category (origin) of the log entry.
+ * @param level      - The level of the log entry.
+ * @param message    - The message of the log entry.
+ * @since
+ */
+export type LoggerCallback2 = (args: LoggerCallbackArgs) => void;
+export type LoggerCallback = LoggerCallback1 | LoggerCallback2;
```

Comment:
To be honest, I'm actually not convinced the change in Core API justifies adding an overload to the callback passed to `setLogger` 🤔 Especially because we need to rely on `arguments.length` to figure out which callback the user has passed ...

Is passing the category as a third positional argument really _that_ awful?

```js
Realm.setLogger((level, message, category) => {
  // ...
});
```